hack/compare_scc.sh

```diff
@@ -19,31 +19,46 @@
 
 set -ex
 
-SCC_BEFORE_SUFFIX=.yaml.before
-SCC_AFTER_SUFFIX=.yaml.after
-
-function dump_sccs_before(){
-    if [ "${CMD}" == "oc" ]; then
-        mkdir -p _out/scc
-        for SCCNAME in $( ${CMD} get scc -o custom-columns=:metadata.name )
-        do
-          echo -e "\n--- SCC ${SCCNAME} ---"
-          ${CMD} get scc ${SCCNAME} -o yaml > "_out/scc/${SCCNAME}${SCC_BEFORE_SUFFIX}" || true
-        done
-    else
-        echo "Ignoring SCCs on k8s"
-    fi
+OUTPUT_DIR=${OUTPUT_DIR:-_out}
+SCC_OUTPUT_DIR="${OUTPUT_DIR}/scc"
+
+BEFORE_DIR="${SCC_OUTPUT_DIR}/before"
+AFTER_DIR="${SCC_OUTPUT_DIR}/after"
+
+SUFFIX=.json
+
+function dump_sccs_before() {
+  if [ "${CMD}" == "oc" ] && [ "${KUBEVIRT_PROVIDER}" != "external" ]; then
+    mkdir -p "${BEFORE_DIR}"
+    for SCCNAME in $( ${CMD} get scc -o custom-columns=:metadata.name); do
+      echo -e "\n--- SCC ${SCCNAME} ---"
+      ${CMD} get scc "${SCCNAME}" -o json | jq 'del(.metadata.resourceVersion,.metadata.generation,.metadata.labels,.metadata.annotations)' > "${BEFORE_DIR}/${SCCNAME}${SUFFIX}" || true
+    done
+  else
+    echo "Ignoring SCCs on k8s"
+  fi
+}
+
+function dump_sccs_after() {
+  if [ "${CMD}" == "oc" ] && [ "${KUBEVIRT_PROVIDER}" != "external" ]; then
+    mkdir -p "${AFTER_DIR}"
+    for SCCNAME in $( ${CMD} get scc -o custom-columns=:metadata.name); do
+      echo -e "\n--- SCC ${SCCNAME} ---"
+      ${CMD} get scc "${SCCNAME}" -o json | jq 'del(.metadata.resourceVersion,.metadata.generation,.metadata.labels,.metadata.annotations)' > "${AFTER_DIR}/${SCCNAME}${SUFFIX}" || true
+    done
+  fi
```

Comment:
the functions `dump_sccs_before()` and `dump_sccs_after()` are very much the same, expect the 1st one stores the manifests on `${BEFORE_DIR}` while the 2nd one on `${AFTER_DIR}`.
I think it can be parametrized and have one dump scc function.

Reply:
Is it blocking?